lib/lists.nix

```diff
@@ -406,50 +842,113 @@ rec {
     else
       genList (n: first + n) (last - first + 1);
 
-  /* Return a list with `n` copies of an element.
+  /**
+    Return a list with `n` copies of an element.
+
+
+    # Inputs
+
+    `n`
+
+    : 1\. Function argument
+
+    `elem`
+
+    : 2\. Function argument
 
-    Type: replicate :: int -> a -> [a]
+    # Type
 
-    Example:
-      replicate 3 "a"
-      => [ "a" "a" "a" ]
-      replicate 2 true
-      => [ true true ]
+    ```
+    replicate :: int -> a -> [a]
+    ```
+
+    # Examples
+    :::{.example}
+    ## `lib.lists.replicate` usage example
+
+    ```nix
+    replicate 3 "a"
+    => [ "a" "a" "a" ]
+    replicate 2 true
+    => [ true true ]
+    ```
+
+    :::
   */
   replicate = n: elem: genList (_: elem) n;
 
-  /* Splits the elements of a list in two lists, `right` and
-     `wrong`, depending on the evaluation of a predicate.
+  /**
+    Splits the elements of a list in two lists, `right` and
+        `wrong`, depending on the evaluation of a predicate.
+
+    # Type
+
+    ```
+    (a -> bool) -> [a] -> { right :: [a]; wrong :: [a]; }
+    ```
+
+    # Examples
+    :::{.example}
+    ## `lib.lists.partition` usage example
+
+    ```nix
+    partition (x: x > 2) [ 5 1 2 3 4 ]
+    => { right = [ 5 3 4 ]; wrong = [ 1 2 ]; }
+    ```
+
+    :::
+  */
+  partition = builtins.partition;
+
+  /**
+    Splits the elements of a list into many lists, using the return value of a predicate.
+        Predicate should return a string which becomes keys of attrset `groupBy` returns.
+
+        `groupBy'` allows to customise the combining function and initial value
```

Comment:
Too much indentation added to subsequent lines - this causes text to be rendered as if it were code.